docs/proposals-accepted/202201-parallel-compaction.md

```diff
@@ -0,0 +1,123 @@
+---
+type: proposal
+title: Compaction of a single tenant in parallel
+status: accepted
+owner: roystchiang
+menu: proposals-accepted
+---
+
+* **Owners:**
+  * @roystchiang
+
+* **Related Tickets:**
+  * [Improvements on Thanos Compactor](https://github.com/thanos-io/thanos/issues/4233)
+
+## Summary
+
+This design document proposes the ability to produce parallelizable compaction plan and the ability to execute them in parallel.
+
+## Motivation
+
+The main motivation for considering parallel compaction in Thanos are the following use cases:
+
+* **A single tenant wants to push more timeseries:** One of the bottlenecks for running a Thanos cluster is how many timeseries can be compacted within the minimal compaction range. Using a compaction range of [2h, 8h, 2d, 25d]  as example, if the compaction for the 2h block is longer than 2 hours, compaction will start to fall behind.
+* **Faster recovery from failed compactor nodes:** In the event of compactor nodes experiencing failures, compaction has to be done for blocks that were uploaded during the outage. By compacting in non-overlapping plans in parallel, we can recover faster.
+
+## Goals
+
+* Ability to create parallelizable compaction plans
+* Ability to execute multiple compaction plans in parallel
```

Comment:
Now a thanos compactor is more than just doing compaction. For other works like downsampling and retention, who will execute those in planner + worker mode?
I think downsampling is very similar to compaction scheduling so maybe the planner can dispatch downsampling job to worker as well? WDYT?

Reply:
I agree that downsampling jobs, progress calculation, and retention should be done by the planner.

The planner has full visibility into the state of the blocks